src/sinks/splunk_hec/common/service.rs

```diff
@@ -44,13 +83,38 @@ impl Service<HecRequest> for HecService {
 
     fn call(&mut self, req: HecRequest) -> Self::Future {
         let mut http_service = self.batch_service.clone();
+        let ack_finalizer_tx = self.ack_finalizer_tx.clone();
         Box::pin(async move {
             http_service.ready().await?;
             let events_count = req.events_count;
             let events_byte_size = req.events_byte_size;
             let response = http_service.call(req).await?;
             let event_status = if response.status().is_success() {
-                EventStatus::Delivered
+                if let Some(ack_finalizer_tx) = ack_finalizer_tx {
+                    let body = serde_json::from_slice::<HecAckResponseBody>(response.body());
+                    match body {
+                        Ok(body) => {
+                            if let Some(ack_id) = body.ack_id {
+                                let (tx, rx) = oneshot::channel();
+                                match ack_finalizer_tx.send((ack_id, tx)) {
+                                    Ok(_) => rx.await.unwrap_or(EventStatus::Rejected),
+                                    // If we cannot send ack ids to the ack client, fall back to default behavior
+                                    Err(_) => EventStatus::Delivered,
```

Comment:
Should we log when this happens? Maybe another internal event?

Reply:
Sure, it could be useful for getting insight if this edge case is encountered.

Reply:
Added an event with a `warning` log: 8d40e3d

Reply:
👍 looks good, thank you!